### PR TITLE
Fix for gcc-8

### PR DESCRIPTION
**Pull request by furushchev**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,10 +2,10 @@ cmake_minimum_required(VERSION 3.5)
 project(wirehair)
 
 set(CMAKE_CXX_STANDARD 11)
-find_program(CXX_10 g++-10)
-if (CXX_10)
-  set(CMAKE_CXX_COMPILER ${CXX_10})
-endif()
+# find_program(CXX_10 g++-10)
```
Comment (garaemon) [inline CMakeLists.txt]: Do you want to comment out this section? If so, let's remove the comment.

File: CMakeLists.txt [hunk 2]
```diff
@@ -61,16 +61,18 @@ set(GEN_TABLES
...
-                    COMMAND sed "s/-march=//g"
-                    COMMAND sed "s/[ \t]*//g"
-                    OUTPUT_VARIABLE DETAILED_ARCHITECTURE
-                    OUTPUT_STRIP_TRAILING_WHITESPACE)
-    set(CMAKE_CXX_FLAGS "-Wall -Wextra -march=${DETAILED_ARCHITECTURE}")
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 9)
+        # Do not use -march=native for distributed compilation.
+        # Run g++ -Q -march=native --help=target and get the native architecture.
+        execute_process(COMMAND g++ -Q -march=native --help=target
+                       COMMAND grep -- -march
```
Comment (garaemon) [inline CMakeLists.txt]: One more space for indent.

File: CMakeLists.txt [hunk 3]
```diff
@@ -61,16 +61,18 @@ set(GEN_TABLES
...
-    # Run g++ -Q -march=native --help=target and get the native architecture.
-    execute_process(COMMAND g++ -Q -march=native --help=target
-                    COMMAND grep -- -march
-                    COMMAND head -n1
-                    COMMAND sed "s/-march=//g"
-                    COMMAND sed "s/[ \t]*//g"
-                    OUTPUT_VARIABLE DETAILED_ARCHITECTURE
-                    OUTPUT_STRIP_TRAILING_WHITESPACE)
-    set(CMAKE_CXX_FLAGS "-Wall -Wextra -march=${DETAILED_ARCHITECTURE}")
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 9)
```
Comment (garaemon) [inline CMakeLists.txt]: Please leave comment that gcc-8 does not support -march=native

File: CMakeLists.txt [hunk 4]
```diff
@@ -61,16 +61,18 @@ set(GEN_TABLES
...
+        # Do not use -march=native for distributed compilation.
+        # Run g++ -Q -march=native --help=target and get the native architecture.
+        execute_process(COMMAND g++ -Q -march=native --help=target
+                       COMMAND grep -- -march
+                       COMMAND head -n1
+                       COMMAND sed "s/-march=//g"
+                       COMMAND sed "s/[ \t]*//g"
+                       OUTPUT_VARIABLE DETAILED_ARCHITECTURE
+                       OUTPUT_STRIP_TRAILING_WHITESPACE)
+        set(CMAKE_CXX_FLAGS "-Wall -Wextra -march=${DETAILED_ARCHITECTURE}")
```
Comment (garaemon) [inline CMakeLists.txt]: I think we should keep `-Wall -Wextra` even if we use gcc-8.

**Review by garaemon — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (furushchev) [reply CMakeLists.txt]: Removed!

File: CMakeLists.txt (hunk 2 above)
Comment (furushchev) [reply CMakeLists.txt]: Oh,,, fixed.

File: CMakeLists.txt (hunk 3 above)
Comment (furushchev) [reply CMakeLists.txt]: Done!

File: CMakeLists.txt (hunk 4 above)
Comment (furushchev) [reply CMakeLists.txt]: Addressed.

**Pull request merged**